app/models/job_proxy_dispatcher.rb

```diff
@@ -144,11 +175,13 @@ def assign_proxy_to_job(proxy, job)
     end
   end
 
-  def pending_jobs
+  def pending_jobs(target_class = VmOrTemplate)
```

Comment:
The only caller of this method is in this file:

`app/models/job_proxy_dispatcher.rb:      jobs_to_dispatch, = Benchmark.realtime_block(:pending_jobs) { pending_jobs }`

If this is not, it should be a private method.